### PR TITLE
Feature/50 navigation

### PR DESCRIPTION
**Pull request by tomklecha**
implementation of navigation component for all features

### CODE REVIEW

File: features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        scenario.onFragment { fragment ->
+            Navigation.setViewNavController(fragment.requireView(), navController)
+        }
+
+        // When
+        onView(ViewMatchers.withId(R.id.questionButton)).perform(ViewActions.click())
+
+        // Then
+        assertEquals(navController.currentDestination?.id, R.id.questionsFragment)
```
Comment (Canato) [inline features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt:51]: declare the variables on the begin
```
val expectedFragment = R.id.questionsFragment
val viewMatcher = ViewMatchers.withId(R.id.questionButton)
val performAction = ViewActions.click()
```

File: features/home/src/main/res/layout/fragment_home.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/fragment_home"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Hello World!"
```
Comment (Canato) [inline features/home/src/main/res/layout/fragment_home.xml:12]: .

File: features/home/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="colorPrimary">#6200EE</color>
+    <color name="colorPrimaryDark">#3700B3</color>
+    <color name="colorAccent">#03DAC5</color>
+</resources>
```
Comment (Canato) [inline features/home/src/main/res/values/colors.xml:6]: Do we need this inside home only? How about we add into common module?

File: features/home/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">Purity Test</string>
+    <string name="login_text">Login</string>
+    <string name="questions_text">Questions</string>
```
Comment (Canato) [inline features/home/src/main/res/values/strings.xml:5]: Common module

**Review by Canato — CHANGES_REQUESTED** [review]
Please return `.idea/codeStyles/` folder so we can enforce the ktlint code style

Thanks for cleaning the rest

- Delete features/home/consumer-rules.pro
- Delete features/home/proguard-rules.pro

Last note, I'm having the feeling that Ktlint didn't check this code 🤔 
Maybe is because you removed codeStyleConfig, many errors, missing lines etc

File: features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomklecha) [reply features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt]: you are right

File: app/src/main/java/com/can_apps/purity_test/MainActivity.kt [hunk 5]
```diff
@@ -1,23 +1,17 @@
...
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        questionButton.setOnClickListener {
-            startActivity(Intent(this, QuestionsActivity::class.java))
-        }
-        loginButton.setOnClickListener {
-            startActivity(Intent(this, LoginActivity::class.java))
-        }
+        val navHostFragment = supportFragmentManager.findFragmentById(R.id.my_nav_fragment) as NavHostFragment?
```
[the remark this answers is not in the document]
Comment (tomklecha) [reply app/src/main/java/com/can_apps/purity_test/MainActivity.kt]: i saw in your project that you were using nullable value, I was confused so I reused it. Change to non null

File: app/src/main/java/com/can_apps/purity_test/MainActivity.kt [hunk 6]
```diff
@@ -1,23 +1,17 @@
...
+        val navController = navHostFragment!!.navController
```
[the remark this answers is not in the document]
Comment (tomklecha) [reply app/src/main/java/com/can_apps/purity_test/MainActivity.kt]: changed

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,41 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (tomklecha) [reply app/src/main/res/layout/activity_main.xml:2]: thank you !

File: features/home/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.can_apps.home">
+
+    /
```
[the remark this answers is not in the document]
Comment (tomklecha) [reply features/home/src/main/AndroidManifest.xml]: done

File: features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.can_apps.home.app.HomeFragment
+import junit.framework.Assert.assertEquals
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class HomeFragmentTest {
```
[the remark this answers is not in the document]
Comment (tomklecha) [reply features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt:21]: please check #53 to resolve it

File: features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt (hunk 1 above)
Comment (tomklecha) [reply features/home/src/androidTest/java/com/can_apps/home/HomeFragmentTest.kt]: changed

File: features/home/src/main/res/layout/fragment_home.xml (hunk 2 above)
Comment (tomklecha) [reply features/home/src/main/res/layout/fragment_home.xml]: shame on me

File: features/home/src/main/res/values/colors.xml (hunk 3 above)
Comment (tomklecha) [reply features/home/src/main/res/values/colors.xml]: was not sure, currently in common

File: features/home/src/main/res/values/strings.xml (hunk 4 above)
Comment (tomklecha) [reply features/home/src/main/res/values/strings.xml]: as above

File: app/src/main/java/com/can_apps/purity_test/MainActivity.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (Canato) [reply app/src/main/java/com/can_apps/purity_test/MainActivity.kt]: I did ask our suggest a change, I make a question =D

I keep my question

[the remark this answers is not in the document]
Comment (tomklecha) [reply app/src/main/java/com/can_apps/purity_test/MainActivity.kt]: ? means that value might be null, so support fragment manager could be null in some case, if maybe view would not be created

File: features/home/src/main/res/layout/fragment_home.xml (hunk 2 above)
Comment (tomklecha) [reply features/home/src/main/res/layout/fragment_home.xml]: can be resolved ?

**Review by Canato — APPROVED** [review]

**Pull request merged**